infrastructure/src/main/java/org/wso2/carbon/testgrid/infrastructure/aws/AWSDeployer.java

```diff
@@ -0,0 +1,190 @@
+import com.amazonaws.services.cloudformation.model.DescribeStacksRequest;
+import com.amazonaws.services.cloudformation.model.DescribeStacksResult;
+import com.amazonaws.services.cloudformation.model.Output;
+import com.amazonaws.services.cloudformation.model.Stack;
+import com.amazonaws.services.cloudformation.model.StackStatus;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.testgrid.common.Deployment;
+import org.wso2.carbon.testgrid.common.Host;
+import org.wso2.carbon.testgrid.common.Infrastructure;
+import org.wso2.carbon.testgrid.common.Script;
+import org.wso2.carbon.testgrid.common.exception.TestGridInfrastructureException;
+import org.wso2.carbon.testgrid.utils.EnvVariableUtil;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is responsible for creating the AWS infrastructure.
+ */
+public class AWSDeployer {
+
+    private Infrastructure infra;
+    private static final Log log = LogFactory.getLog(AWSDeployer.class);
+
+    /**
+     * This constructor creates AWS deployer object and validate AWS related environment variables are present.
+     *
+     * @param awsKeyVariableName    Environment variable name for AWS ACCESS KEY.
+     * @param awsSecretVariableName Environment variable name for AWS SECRET KEY.
+     * @throws TestGridInfrastructureException Throws exception when environment variables are not set.
+     */
+    public AWSDeployer(String awsKeyVariableName, String awsSecretVariableName) throws TestGridInfrastructureException {
+        String awsIdentity = EnvVariableUtil.readEnvironmentVariable(awsKeyVariableName);
+        String awsSecret = EnvVariableUtil.readEnvironmentVariable(awsSecretVariableName);
+        if ((awsIdentity == null) || (awsSecret == null)) {
+            String errorMessage = "AWS Credentials must be set as environment variables";
+            throw new TestGridInfrastructureException(errorMessage);
+        }
+    }
+
+    /**
+     * This method initiates creating infrastructure through CLoudFormation.
+     *
+     * @param script       Script object containing the CF details.
+     * @param infraRepoDir Path of TestGrid home location in file system as a String.
+     * @return a Deployment object with created infrastructure details.
+     * @throws InterruptedException            when an Interrupt occurs while waiting for CF result.
+     * @throws IOException                     when an error occurs while reading the cf template file.
+     * @throws TestGridInfrastructureException AWS related error caused by malformed CF template.
+     */
+    public Deployment createInfrastructure(Script script, String infraRepoDir) throws InterruptedException,
+            IOException, TestGridInfrastructureException {
+        String cfName = script.getName();
```

Comment:
Better rename the variable to `cloudFormationName` instead of `cfName` since it would indicate the purpose of this variable